### PR TITLE
feat: add AWS GOV support in api_integration

### PR DESCRIPTION
**Pull request by sfc-gh-kmaurya**
Added aws_gov_api_gateway and aws_gov_private_api_gateway in api_provider list in api_integration resource
## Test Plan

* [x] careful review

## References

* API Integration [doc](https://docs.snowflake.com/en/sql-reference/sql/create-api-integration.html)
* issue ref #1113

### CODE REVIEW

File: pkg/resources/api_integration_test.go [hunk 1]
```diff
@@ -24,15 +24,15 @@ func TestAPIIntegrationCreate(t *testing.T) {
...
+		"api_provider":         "aws_gov_api_gateway",
 		"api_aws_role_arn":     "arn:aws:iam::000000000001:/role/test",
 	}
 	d := schema.TestResourceDataRaw(t, resources.APIIntegration().Schema, in)
 	r.NotNil(d)
 
 	WithMockDb(t, func(db *sql.DB, mock sqlmock.Sqlmock) {
 		mock.ExpectExec(
-			`^CREATE API INTEGRATION "test_api_integration" API_PROVIDER=aws_api_gateway API_AWS_ROLE_ARN='arn:aws:iam::000000000001:/role/test' API_ALLOWED_PREFIXES=\('https://123456.execute-api.us-west-2.amazonaws.com/prod/'\) ENABLED=true$`,
+			`^CREATE API INTEGRATION "test_api_integration" API_PROVIDER=aws_gov_api_gateway API_AWS_ROLE_ARN='arn:aws:iam::000000000001:/role/test' API_ALLOWED_PREFIXES=\('https://123456.execute-api.us-west-2.amazonaws.com/prod/'\) ENABLED=true$`,
```
Comment (sfc-gh-jalin) [inline pkg/resources/api_integration_test.go:35]: Should we create a new test rather than overwrite an existing test?

Comment (sfc-gh-kmaurya) [reply pkg/resources/api_integration_test.go:35]: Yes sure, I will create new test.

**Review by sfc-gh-afedorov — APPROVED** [review]
lgtm, but remove the tests because we don't have a GovCloud account to test on (we need this because we're building one!)

**Pull request merged**